gtsam/navigation/tests/testAHRSFactor.cpp

```diff
@@ -462,46 +367,115 @@ TEST (AHRSFactor, predictTest) {
 
   // Actual Jacobians
   Matrix H;
-  (void) pim.predict(bias,H);
+  (void)pim.predict(bias, H);
   EXPECT(assert_equal(expectedH, H, 1e-8));
 }
 //******************************************************************************
-#include <gtsam/linear/GaussianFactorGraph.h>
-#include <gtsam/nonlinear/Marginals.h>
-
-TEST (AHRSFactor, graphTest) {
+TEST(AHRSFactor, graphTest) {
   // linearization point
-  Rot3 x1(Rot3::RzRyRx(0, 0, 0));
-  Rot3 x2(Rot3::RzRyRx(0, M_PI / 4, 0));
-  Vector3 bias(0,0,0);
+  Rot3 Ri(Rot3::RzRyRx(0, 0, 0));
+  Rot3 Rj(Rot3::RzRyRx(0, M_PI / 4, 0));
+  Vector3 bias(0, 0, 0);
 
   // PreIntegrator
   Vector3 biasHat(0, 0, 0);
-  PreintegratedAhrsMeasurements pim(biasHat, kMeasuredAccCovariance);
+  PreintegratedAhrsMeasurements pim(biasHat, kMeasuredOmegaCovariance);
 
   // Pre-integrate measurements
   Vector3 measuredOmega(0, M_PI / 20, 0);
   double deltaT = 1;
 
   // Create Factor
-  noiseModel::Base::shared_ptr model = //
+  noiseModel::Base::shared_ptr model =  //
       noiseModel::Gaussian::Covariance(pim.preintMeasCov());
   NonlinearFactorGraph graph;
   Values values;
   for (size_t i = 0; i < 5; ++i) {
     pim.integrateMeasurement(measuredOmega, deltaT);
   }
 
-  // pim.print("Pre integrated measurementes");
-  AHRSFactor factor(X(1), X(2), B(1), pim, kZeroOmegaCoriolis);
-  values.insert(X(1), x1);
-  values.insert(X(2), x2);
+  // pim.print("Pre integrated measurements");
+  AHRSFactor factor(R(1), R(2), B(1), pim, kZeroOmegaCoriolis);
+  values.insert(R(1), Ri);
+  values.insert(R(2), Rj);
   values.insert(B(1), bias);
   graph.push_back(factor);
   LevenbergMarquardtOptimizer optimizer(graph, values);
   Values result = optimizer.optimize();
   Rot3 expectedRot(Rot3::RzRyRx(0, M_PI / 4, 0));
-  EXPECT(assert_equal(expectedRot, result.at<Rot3>(X(2))));
+  EXPECT(assert_equal(expectedRot, result.at<Rot3>(R(2))));
+}
+
+/* ************************************************************************* */
+TEST(AHRSFactor, bodyPSensorWithBias) {
```

Comment:
This is the new end-to-end test, with sensor rotation and large bias. No aiding, so prior on bias makes solution observable.